---
title: "TRNValStandVisVignettes"
author: "Mogtaba Alim"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{TRNValStandVisVignettes}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# TRNValStandVis

## Introduction

`TRNValStandVis` is an R packacge that validates and 
It combines transcription factor - target gene interaction data from four data sets, 
the Molecular Signatures Database (MSigDB) by GSEA, hTFtarget by Zhang et al.,
the ChIP-X Enrichment Analysis Version 3 (ChEA3) by Ma'ayan lab, and TFLink by Liska et al. to 
create a single combined dataset, where all TF's and TG's are standardized to be HGNC compliant. 
The combined data was used to create a qualitative score to evaluate the quality of TF-TG interactions
as well a score to assess the overall quality of a specific TF. Further abilities include
visualizing TF-TG regulons and networks and comparing different TF's qualitatively. 
This package will allow for the validation of TF-TG's across different sources, 
quality metrics in regards to level of validation for each TF-TG, as well as 
standardization in regards to nomenclature.

Use the following commands to download **TRNValStandVis**

```{r, eval= FALSE}
install.packages("devtools")
library("devtools")
devtools::install_github("Mogtaba-Alim/TRNValStandVis", build_vignettes = TRUE)
library("TRNValStandVis")
```

To list all sample functions available in the package:

```{r, eval= FALSE}
 ls("package:TRNValStandVis")
```
 
 TO list all the sample datasets in the package:
 
```{r, eval= FALSE}
data(package= "TRNValStandVis")
```
 
### Minimal Requirements
 
 The minimal requirements for this package include, the R. directory, data which has
 the packages data, Description and Namespace. 
 
### Applications 
 
 
1. get_TGs

-Take in a TF and return all its target genes. Can be adjusted based on number of datasets TF-TG appears in.

```{r, eval= FALSE}
library(TRNValStandVis)
?TRNValStandVis::get_TGs
```

2. get_TFs

- Take in a gene and return all the transcription factors that regulate it. Can be adjusted based on number of datasets TF-TG appears in.

```{r, eval= FALSE}
library(TRNValStandVis)
?TRNValStandVis::get_TFs
```

3. plot_TF_regulon

- Take in a gene and plot that gene’s regulon including regulators and targets.

```{r, eval= FALSE}
library(TRNValStandVis)
?TRNValStandVis::plot_TF_regulon
```

4. plot_TFs_network

- Take in up to 3 gene names and plot the network involving all the genes regulons combined.

```{r, eval= FALSE}
library(TRNValStandVis)
?TRNValStandVis::plot_TFs_network
```

5. get_TF_TG_val

- Take in a TF and return the validity score for all the TG’s of that TF

```{r, eval=FALSE}
library(TRNValStandVis)
?TRNValStandVis::get_TF_TG_val
```

6. plot_TF_TG_val

- Take in a TF and plot the validity score for all the TG’s of that TF

```{r, eval= FALSE}
library(TRNValStandVis)
?TRNValStandVis::plot_TF_TG_val
```

7. get_TF_total_val

- Takes in a TF and provides the average validty score for that TF

```{r, eval= FALSE}
library(TRNValStandVis)
?TRNValStandVis::get_TF_total_val
```

8. plot_TF_total_val

- Take in a collection of TF names and plots their average validity score

```{r, eval= FALSE}
library(TRNValStandVis)
?TRNValStandVis::plot_TF_total_val
```

## Package References

Dplyr. (n.d.). Retrieved November 15, 2023, from Tidyverse.org website: https://dplyr.tidyverse.org/

Fujiyoshi, K., Bruford, E. A., Mroz, P., Sims, C. L., O’Leary, T. J., Lo, A. W. I., … Ogino, S. (2021). Standardizing gene product nomenclature—a call to action. Proceedings of the National Academy of Sciences of the United States of America, 118(3). doi:10.1073/pnas.2025207118

Ggplot2. (n.d.). Retrieved November 15, 2023, from Tidyverse.org website: https://ggplot2.tidyverse.org

igraph – Network analysis software. (n.d.). Retrieved November 15, 2023, from Igraph.org website: https://igraph.org/

Lachmann, A., Xu, H., Krishnan, J., Berger, S. I., Mazloom, A. R., & Ma’ayan, A. (2010). ChEA: transcription factor regulation inferred from integrating genome-wide ChIP-X experiments. Bioinformatics (Oxford, England), 26(19), 2438–2444. doi:10.1093/bioinformatics/btq466

Liberzon, A., Birger, C., Thorvaldsdóttir, H., Ghandi, M., Mesirov, J. P., & Tamayo, P. (2015). The molecular signatures database hallmark gene set collection. Cell Systems, 1(6), 417–425. doi:10.1016/j.cels.2015.12.004

Liska, O., Bohár, B., Hidas, A., Korcsmáros, T., Papp, B., Fazekas, D., & Ari, E. (2022). TFLink: an integrated gateway to access transcription factor–target gene interactions for multiple species. Database: The Journal of Biological Databases and Curation, 2022, baac083. doi:10.1093/database/baac083

The R project for statistical computing. (n.d.). Retrieved November 15, 2023, from R-project.org website: http://www.R-project.org

Zhang, Q., Liu, W., Zhang, H.-M., Xie, G.-Y., Miao, Y.-R., Xia, M., & Guo, A.-Y. (2020). HTFtarget: A comprehensive database for regulations of human transcription factors and their targets. Genomics, Proteomics & Bioinformatics, 18(2), 120–128. doi:10.1016/j.gpb.2019.09.006


